src/mrpro/algorithms/optimizers/_fista.py

```diff
@@ -0,0 +1,204 @@
+"""FISTA algorithm."""
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import math
+
+import torch
+
+from mrpro.operators._Functional import Functional
+
+
+def grad(f: Functional, x: torch.Tensor):
+    """Compute the gradient of f(x).
+
+    Parameters
+    ----------
+    f
+        differentiable function
+    x
+        point in the domain of f
+    """
+    x = x.clone().detach().requires_grad_(True)
+    y = f(x)
```

Comment:
I believe it should be `y, = f(x)`

Reply:
why?

Reply:
I think the functionals are operators, thus returning a tuple of tensors (even though it is a single tensor)? 
Or does it work as is? If yes, I am surprised and wondering why and how..

Reply:
no it was just my curiosity! I forgot they returned tuple